src/main/thrift/parquet.thrift

```diff
@@ -162,19 +162,48 @@ enum ConvertedType {
   BSON = 20;
 
   /**
+   * @Deprecated: use INTERVAL_YEAR_MONTH or INTERVAL_DAY_TIME
+   * since the SQL standard defines either YEAR_MONTH or DAY_TIME unit.
+   * This is deprecated in favor of those 2 types
```

Comment:
@rdblue, this is the complete paragraph defining the comparison of day-time intervals where the quote was taken from.

         Day-time intervals are mutually comparable only with other day-
         time intervals. If two day-time intervals have different interval
         precisions, they are, for the purpose of any operations between
         them, effectively converted to the same precision by appending new
         <datetime field>s to either the most significant end or the least
         significant end of one or both day-time intervals. New least sig-
         nificant <datetime field>s are assigned a value of 0. When it is
         necessary to add new most significant datetime fields, the asso-
         ciated value is effectively converted to the new precision in a
         manner obeying the natural rules for dates and times associated
         with the Gregorian calendar.

To my understanding, this means that, for example, comparing `INTERVAL '100' HOUR` with `INTERVAL '4 04:00' DAY TO MINUTE` requires conversion to the common super type, in this case `INTERVAL DAY TO MINUTE`. During conversion least significant fields are filled with zeros and most significant "in a manner obeying the natural rules for dates and times associated with the Gregorian calendar", since the non-most-significant fields must conform to the constraints in the table (comes from the same chapter). In this case `INTERVAL '100' HOUR` becomes `INTERVAL '4 04:00' DAY TO MINUTE`. Now they are compared field by field (making them equal).
This is the best reference from the SQL '92 standard I can give to argument 1 day = 24 hours. And while implementations of the standard vary they still tend to use the above method for comparisons.